pkg/util/util.go

```diff
@@ -232,58 +235,50 @@ func ParseMachinesYaml(file string) ([]*clusterv1.Machine, error) {
 	defer reader.Close()
 
 	decoder := yaml.NewYAMLOrJSONDecoder(reader, 32)
-	machineList, err := decodeMachineLists(decoder)
 
-	if err != nil {
+	var (
+		bytes       [][]byte
+		machineList clusterv1.MachineList
+		machine     clusterv1.Machine
+		machines    = []clusterv1.Machine{}
+	)
+
+	// TODO: use the universal decoder instead of doing this.
+	if bytes, err = decodeClusterV1Kinds(decoder, "MachineList"); err != nil {
+		if isMissingKind(err) {
+			err = fmt.Errorf(MachineListFormatDeprecationMessage)
```

Comment:
we try to use https://github.com/pkg/errors
has a stack-trace.

`errors.New()`
[1]

Reply:
👍